### PR TITLE
fixed flaky test TestAccSecurityCenterManagement

### PR DESCRIPTION
**Pull request by vijaykanthm**
This PR aims to fix the ETD Module failing test issue raised in #[19086](https://github.com/hashicorp/terraform-provider-google/issues/19086) 

References to the previous PR's worked on this: 
#[11469](https://github.com/GoogleCloudPlatform/magic-modules/pull/11469) & [#11650](https://github.com/GoogleCloudPlatform/magic-modules/pull/11650)
ETD Custom Module REST Doc link: https://cloud.google.com/security-command-center/docs/reference/security-center-management/rest/v1/organizations.locations.eventThreatDetectionCustomModules 

**Release Note Template for Downstream PRs (will be copied)**

```release-note:none
securitycentermanagement: fixed flaky test TestAccSecurityCenterManagement
```

### CODE REVIEW

**Comment by zli82016** [comment]
There are four test cases inside `func TestAccSecurityCenterManagement`. 
The version guard is needed for this function and the four test cases.

That means for all of the four test files in the directory `mmv1/third_party/terraform/services/securitycentermanagement`, the version guard (`<% unless version == 'ga' -%>`) is needed at the top of the page. 

[Example](https://github.com/GoogleCloudPlatform/magic-modules/blob/main/mmv1/third_party/terraform/services/apigateway/resource_api_gateway_api_config_test.go.erb)

File: mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb [hunk 1]
```diff
@@ -90,12 +93,18 @@ resource "google_scc_management_organization_event_threat_detection_custom_modul
 			  ]}
             EOF
 }
+<% else -%>
+# GA version doesn't support this field
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb:97]: The else block is not needed.

File: mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb [hunk 2]
```diff
@@ -112,6 +121,9 @@ resource "google_scc_management_organization_event_threat_detection_custom_modul
...
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb:125]: The else block is not needed.

**Review by zli82016 — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb (hunk 2 above)
Comment (vijaykanthm) [reply mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb]: Removed.

File: mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb [hunk 3]
```diff
@@ -90,12 +93,16 @@ resource "google_scc_management_organization_event_threat_detection_custom_modul
...
 }
+<% end -%>
 `, context)
 }
 
 func testAccSecurityCenterManagementOrganizationEventThreatDetectionCustomModule_sccOrganizationCustomModuleUpdate(context map[string]interface{}) string {
 	return acctest.Nprintf(`
+<% unless version == 'ga' -%>
 resource "google_scc_management_organization_event_threat_detection_custom_module" "example" {
+	provider = google-beta
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb:104]: `provider = google-beta` need to be added to the tests in the other three files as well.

File: mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb [hunk 4]
```diff
@@ -45,7 +45,7 @@ func testAccSecurityCenterManagementOrganizationEventThreatDetectionCustomModule
 
 	acctest.VcrTest(t, resource.TestCase{
 		PreCheck:                 func() { acctest.AccTestPreCheck(t) },
-		ProtoV5ProviderFactories: acctest.ProtoV5ProviderFactories(t),
+		ProtoV5ProviderFactories: acctest.ProtoV5ProviderBetaFactories(t), // Ensure beta factory is used
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb:48]: `acctest.ProtoV5ProviderBetaFactories` need to be used to the tests in the other three files as well.

**Review by zli82016 — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb [hunk 5]
```diff
@@ -72,7 +72,10 @@ func testAccSecurityCenterManagementOrganizationEventThreatDetectionCustomModule
 
 func testAccSecurityCenterManagementOrganizationEventThreatDetectionCustomModule__sccOrganizationCustomModuleExample(context map[string]interface{}) string {
 	return acctest.Nprintf(`
+<% unless version == 'ga' -%>
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb:75]: Can you move `<% unless version == 'ga' -%>` to the top of this file as what is done for the other three files?

**Review by zli82016 — CHANGES_REQUESTED** [review]

File: mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb [hunk 6]
```diff
@@ -112,6 +119,7 @@ resource "google_scc_management_organization_event_threat_detection_custom_modul
 			  ]}
             EOF
 }
+<% end -%>
```
Comment (zli82016) [inline mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb:122]: `<% end -%>` needs to be moved to the bottom.

**Comment by c2thorn** [comment]
Hi @vijaykanthm 

Tomorrow, the Magic Modules repository is scheduled to undergo a language migration from Ruby to Go. You can view more details about this in our announcement here: https://github.com/hashicorp/terraform-provider-google/issues/19583 (or [go/mm-migration-announcement](go/mm-migration-announcement) if you are a Googler)

This open pull request may become incompatible due to most YAML and .erb files converting to Go-compatible files.

Our team (Magic Modules repository maintainers) has tooling to automatically convert changes to the new language, and we can prepare a new commit for this pull request that is compatible with the migration.

In order to push the new changes to your pull request, we would need to <ins>force push the commit to your fork's branch</ins>. Our tooling saves a backup branch before converting, so we could rollback or open a new pull request if needed. We would also work with you and the PR reviewer in the event additional changes are needed.

You also have the option to update the pull request yourself after the migration. You can view a [preview branch](https://github.com/GoogleCloudPlatform/magic-modules/tree/FEATURE-BRANCH-rewrite-switchover-preview) and [updated documentation](https://googlecloudplatform.github.io/magic-modules/reference/ruby-go-changes/) related to the migration changes.

We will take no action until we have your explicit permission to push changes to your fork's branch used for this pull request. Let me or your reviewer know if you have any further questions!

File: mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb (hunk 3 above)
Comment (vijaykanthm) [reply mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb:102]: Addressed.

File: mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb (hunk 4 above)
Comment (vijaykanthm) [reply mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb:49]: Addressed.

File: mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb (hunk 5 above)
Comment (vijaykanthm) [reply mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb]: Addressed.

File: mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb (hunk 6 above)
Comment (vijaykanthm) [reply mmv1/third_party/terraform/services/securitycentermanagement/resource_scc_management_organization_event_threat_detection_custom_module_test.go.erb]: Addressed.

**Review by zli82016 — APPROVED** [review]
LGTM

**Pull request merged**